### PR TITLE
PYTHON-2797 Update changelog for 5.0 support

### PR DESCRIPTION
**Pull request by ShaneHarvey**

### CODE REVIEW

File: doc/changelog.rst [hunk 1]
```diff
@@ -16,20 +16,39 @@ Changes in Version 3.12.0
...
+- Support for connecting to load balanced MongoDB clusters via the new
+  ``loadBalanced`` URI option.
 - Added :attr:`pymongo.mongo_client.MongoClient.topology_description`.
 - Added hash support to :class:`~pymongo.mongo_client.MongoClient`,
   :class:`~pymongo.database.Database` and
   :class:`~pymongo.collection.Collection` (`PYTHON-2466`_).
 - Improved the error message returned by
   :meth:`~pymongo.collection.Collection.insert_many` when supplied with an
   argument of incorrect type (`PYTHON-1690`_).
+- Add session support to :meth:`~pymongo.collection.Collection.find_raw_batches`
```
Comment (prashantmital) [inline doc/changelog.rst:37]: Also read concern support? (PYTHON-2734)

File: doc/changelog.rst [hunk 2]
```diff
@@ -16,20 +16,39 @@ Changes in Version 3.12.0
...
 
 - Fixed a bug that could cause the driver to deadlock during automatic
   client side field level encryption (`PYTHON-2472`_).
+- Fixed a potential deadlock when garbage collecting an unclosed exhaust
+  :class:`~pymongo.cursor.Cursor`.
+- Fixed an bug where using gevent.Timeout to timeout an operation could
+  lead to a deadlock.
+- Fixed the following bug with Atlas Data Lake. When closing cursors,
+  pymongo now sends killCursors with the namespace returned the cursor's
+  initial command response.
```
Comment (prashantmital) [inline doc/changelog.rst:52]: Missed bugfix: 
```
- Fixed a bug in :class:`~pymongo.cursor.RawBatchCursor` that caused it to return an empty bytestring when the cursor contained no results. It now raises :exc:`StopIteration` instead.
```

File: doc/changelog.rst [hunk 3]
```diff
@@ -16,20 +16,39 @@ Changes in Version 3.12.0
 Notable improvements
 ....................
 
+- Added support for MongoDB 5.0.
 - Support for MongoDB Versioned API, see :class:`~pymongo.server_api.ServerApi`.
+- Support for snapshot reads on secondaries (see :ref:`snapshot-reads-ref`).
```
Comment (prashantmital) [inline doc/changelog.rst:21]: Do we want to mention timeseries support and link to `create_collection`?

Comment (prashantmital) [reply doc/changelog.rst:21]: Same for dots and dollars?

**Review by prashantmital — CHANGES_REQUESTED** [review]
See comments inline.

**Comment by ShaneHarvey** [comment]
Updated.

**Review by prashantmital — APPROVED** [review]

**Pull request merged**